### PR TITLE
Remove build-time openssh-server host keys and add service to auto-generating them on boot.

### PR DESCRIPTION
**Pull request by sil2100**
This is apparently another thing that we missed because of not looking at livecd-rootfs code and the core-builder parts. Currently we seem to ship ssh host keys which, as I understand it (and see it) causes each image have the same ssh key fingerprint (!).

This is all pulled in from live-build's remove-openssh-server-host-keys hook and the files in core-build repository.

### CODE REVIEW

**Comment by sil2100** [comment]
Tested this and it seems to work properly now. Previously every new image built had the same SSH host key everytime. Now each one has a new, 'unique' ssh host key generated on boot as expected.

**Comment by sil2100** [comment]
mvo said it looks ok so I'll just 'do it' and merge.

File: static/lib/systemd/system/snapd.sshd-keygen.service [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[Unit]
+Description=Generate sshd host keys
+Before=sshd.service
```
Comment (zyga) [inline static/lib/systemd/system/snapd.sshd-keygen.service:3]: I would much rather see an either:
 - first boot only service
 - if-not-exists service

File: static/usr/lib/snapd/sshd-host-keygen [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+set -eu
+
+# taken from the openssh-server.postinst
+get_config_option() {
+	option="$1"
+
+	[ -f /etc/ssh/sshd_config ] || return
+
+	# TODO: actually only one '=' allowed after option
+	perl -lne '
```
Comment (zyga) [inline static/usr/lib/snapd/sshd-host-keygen:12]: I would love to avoid using perl here.

File: static/usr/lib/snapd/sshd-host-keygen [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+
```
Comment (zyga) [inline static/usr/lib/snapd/sshd-host-keygen:2]: This unit needs to be documented. It should say how the unit relates to ssh configuration.

File: static/usr/lib/snapd/sshd-host-keygen [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	shift
+	file="$1"
+	shift
+
+	if echo "$hostkeys" | grep -x "$file" >/dev/null && \
+	   [ ! -f "$file" ] ; then
+		printf "%s" "$msg"
+		ssh-keygen -q -f "$file" -N '' "$@"
+		echo
+		if which restorecon >/dev/null 2>&1; then
```
Comment (zyga) [inline static/usr/lib/snapd/sshd-host-keygen:45]: Do we need to restore selinux context? This feels like it was copy-pasted from somewhere generic.

File: static/usr/lib/snapd/sshd-host-keygen [hunk 5]
```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
```
Comment (zyga) [inline static/usr/lib/snapd/sshd-host-keygen:1]: Is there something that runs shellcheck against this script?

**Review by zyga — COMMENTED** [review]
I was pinged about this pull request but I only noticed it after the fact it was merged. Sending a post-merge review. I would be -1 on the pull request in the original state.

**Pull request merged**